flake.nix

```diff
@@ -0,0 +1,151 @@
+      riscv64-unknown-none-elf-stdenv = pkgs.stdenv.override {
+        targetPlatform = {
+          config = "riscv64-unknown-elf";
+          system = "riscv64-none";
+          parsed = {
+            kernel = {
+              execFormat = {
+                name = "unknown";
+              };
+              name = "none";
+            };
+          };
+          isiOS = false;
+          isAarch32 = false;
+          isWindows = false;
+          isMips64n64 = false;
+          isMusl = false;
+          isPower = false;
+          isVc4 = false;
+        };
+      };
+      riscv32-unknown-none-elf-stdenv = pkgs.stdenv.override {
+        targetPlatform = {
+          config = "riscv32-unknown-elf";
+          system = "riscv32-none";
+          parsed = {
+            kernel = {
+              execFormat = {
+                name = "unknown";
+              };
+              name = "none";
+            };
+          };
+          isiOS = false;
+          isAarch32 = false;
+          isWindows = false;
+          isMips64n64 = false;
+          isMusl = false;
+          isPower = false;
+          isVc4 = false;
+        };
+      };
+
+      rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+
+      #since hubris looks for 64bit named binaries
+      binutils32 = pkgs.binutils-unwrapped.override {
+        stdenv = riscv32-unknown-none-elf-stdenv;
+      };
+      binutils64 = pkgs.binutils-unwrapped.override {
+        stdenv = riscv64-unknown-none-elf-stdenv;
+      };
+
+      hubris = (
+        {
+          app,
+          toml,
+          doCheck ? false,
+        }:
+          pkgs.callPackage ./nix/hubris.nix {
+            inherit binutils64 app toml doCheck;
+            cargo = rust;
+            rustc = rust;
+            src = pkgs.lib.cleanSource ./.;
+            version = "beta";
+          }
+      );
+
+      hubris-test-suite-runner = (
+        {
+          hubris,
+          app,
+        }:
+          pkgs.callPackage ./nix/qemu-test-suite.nix {
+            inherit hubris app;
+            humility = humilityflake.packages.${system}.humility;
+          }
+      );
+
+      target = "demo-hifive-inventor";
+    in {
+      packages = flake-utils.lib.flattenTree {
+        demo-hifive1-revb = hubris {
+          app = "demo-hifive1-revb";
+          toml = "app/demo-hifive1-revb/app.toml";
+        };
+        demo-hifive-inventor = hubris {
+          app = "demo-hifive-inventor";
+          toml = "app/demo-hifive-inventor/app.toml";
+        };
+        tests-hifive-inventor = hubris {
+          app = "tests-hifive-inventor";
+          toml = "test/tests-hifive-inventor/app.toml";
+        };
+      };
+
+      devShells.default = pkgs.mkShell {
+        shellHook = ''
+          export HUMILITY_ARCHIVE=$(pwd)/target/${target}/dist/default/build-${target}.zip
+          export CARGO_HOME=$HOME/.cargo
+        '';
+
+        nativeBuildInputs = with pkgs; [
+          rust
+          qemu
```

Comment:
I'm assuming this is where you specify that you want QEMU in the environment? If so, how do we know that this pulls RISC-V QEMU? Does it just pull all of them?

Reply:
Just looked in my Nix shell, seems like it got all of them. Is there any way to restrict it to just the riscv binaries?

Reply:
It pulls all of them. If you just want the riscv binaries it would trigger a rebuild of qemu, rather than pull from binary caches.

Reply:
I see. That's a bit annoying as there's a bunch of bloat, but it's better than requiring a full rebuild.